scripts/actions/fetch-observability-packs.js

```diff
@@ -0,0 +1,115 @@
+'use strict';
+
+/**
+ * This script is used to query the New Relic GraphQL API for Observability Packs.
+ * It then writes the array of Observability Packs to src/data/observability-packs.json
+ * It requires the following environment variables to be set:
+ * NR_GQL_URL - The New Relic GraphQL URL
+ * NR_API_TOKEN - A New Relic personal API token
+ **/
+
+/* eslint-disable no-console */
+const fs = require('fs');
+const fetch = require('node-fetch');
+
+const PACKS_FILE_PATH = './src/data/observability-packs.json';
+const NR_GQL_URL = process.env.NR_GQL_URL;
+const NR_API_TOKEN = process.env.NR_API_TOKEN;
+
+const packQuery = `# gql 
+  {
+    docs {
+      openInstallation {
+        observabilityPackSearch {
+          count
+          results {
+            observabilityPacks {
+              authors
+              dashboards {
+                description
+                name
+                screenshots
+                url
+              }
+              description
+              iconUrl
+              id
+              level
+              logoUrl
+              name
+              websiteUrl
+            }
+          }
+        }
+      }
+    }
+  }
+`;
+
+/**
+ * Queries graphql for the provided query
+ * @param {String} queryString the graphql query to send
+ * @param {String} url NR graphql endpoint
+ * @param {String} token NR api token
+ * @returns {Promise<Object[]|undefined>} returns the resulting array
+ * or `undefined` if there was an error
+ **/
+const fetchPacks = async (queryString, url, token) => {
+  try {
+    const results = await fetch(url, {
+      method: 'post',
+      body: JSON.stringify({ query: queryString }),
+      headers: {
+        'Content-Type': 'application/json',
+        'Api-Key': token,
+      },
+    }).then((res) => res.json());
+
+    if (results.errors) {
+      console.log('GRAPHQL Errors:', JSON.stringify(results.errors, null, 2));
+    }
```

Comment:
If there are `results.errors` we probably want to drop out of this function (either by `return` or `throw`ing).

Reply:
My understanding of GraphQL is that it can still return partial information even if part of the query errors out. The API currently returns errors for the description field being blank on dashboards, but still returns the rest of the fields. Do you think it should just error out in that case?

Reply:
If the API changes and this query is no longer accurate, then we run the risk of hitting a hard-to-track-down error. For example, this query will return errors and _not_ contain `data.docs`:

```
{
  docs {
    openInstallation {
      observabilityPackSearch {
        results {
          observabilityPacks {
            taco
          }
        }
      }
    }
  }
}
```

I suppose it's not _critical_ that we throw an error here. If we don't then we _should_ consider checking the properties exist before returning (the line below this).